pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -135,15 +200,31 @@ def __init__(
         c_dtype=np.complex128,
         shots=None,
         batch_obs: Union[bool, int] = False,
+        analytic=None,
     ):
-        super().__init__(wires, c_dtype=c_dtype, shots=shots)
-        self._gpu_state = _gpu_dtype(self._state.dtype)(self._state)
+        if c_dtype is np.complex64:
+            r_dtype = np.float32
+            self.use_csingle = True
+        elif c_dtype is np.complex128:
+            r_dtype = np.float64
+            self.use_csingle = False
+        else:
+            raise TypeError(f"Unsupported complex Type: {c_dtype}")
+        super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype, analytic=analytic)
+        self._gpu_state = _gpu_dtype(c_dtype)(self.num_wires)
+        self._create_basis_state_GPU(0)
         self._sync = sync
         self._dp = DevPool()
         self._batch_obs = batch_obs
 
+        self._state = self._create_basis_state_default(0)
+        self._pre_rotated_state = self._state
+
     def reset(self):
         super().reset()
+        # init the state vector to |00..0>
+        self._state = self._create_basis_state_default(0)
+        self._pre_rotated_state = self._state
         self._gpu_state.resetGPU(False)  # Sync reset
```

Comment:
If we reset the GPU this will always create a CPU-backed statevector, right?

Reply:
lines 225-257 removed!